[SYCL][COMPAT] Added filter_device and list_devices (#14016)
intel/llvm: sycl/doc/syclcompat/README.md

// Util function to get a CPU device.
static inline device_ext &cpu_device();

/// Filter out devices; only keep the device whose name contains one of the
/// subname in \p dev_subnames.
/// May break device id mapping and change current device. It's better to be
/// called before other SYCLcompat or SYCL APIs.
static inline void filter_device(const std::vector<std::string> &dev_subnames);

> /// List all the devices with its id in dev_mgr.

Reviewer: ```suggestion
/// Print all the devices (and their IDs) in the dev_mgr
```

throws a `sycl::exception` if the device does not have the specified list of
`sycl::aspect`.

Devices can be listed and filtered using `syclcompat::list_devices()` and
`syclcompat::filter_device()`. If `SYCLCOMPAT_VERBOSE` is defined at compile
time, the available SYCL devices are printed to the standard output both at

> initialization time, or if the device list is filtered using

Reviewer: ```suggestion
initialization time, and when the device list is filtered using
```
